### PR TITLE
[ENHANCEMENT] Promtool: Adding labels to time series while creating tsdb blocks

### LINKED ISSUE #14402: [Feature] Add new labels to time series
### Proposal

While using `promtool tsdb create-blocks-from openmetrics` to create a tsdb blocks there should be a way to add custom labels to time series.

### PR DESCRIPTION
**Pull request by patilsuraj767**
Fixes #14402 

Usage - 
```
promtool tsdb create-blocks-from openmetrics --label=cluster_name=demo --label=user=test /path/to/file
```

### CODE REVIEW

**Review by machine424 — COMMENTED** [review]
Thanks for this, it seems legitimate to me, left some comments.

Maybe we need to add a warning if a label is overridden...?

**Comment by dgl** [comment]
@patilsuraj767 thanks for the contribution, would you be able to add some tests for this functionality?

File: cmd/promtool/backfill.go [hunk 1]
```diff
@@ -162,7 +162,13 @@ func createBlocks(input []byte, mint, maxt, maxBlockDuration int64, maxSamplesIn
 				l := labels.Labels{}
 				p.Metric(&l)
 
-				if _, err := app.Append(0, l, *ts, v); err != nil {
+				lb := labels.NewBuilder(l)
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply cmd/promtool/backfill.go:165]: ok

File: cmd/promtool/main.go [hunk 2]
```diff
@@ -252,6 +252,7 @@ func main() {
 	importQuiet := importCmd.Flag("quiet", "Do not print created blocks.").Short('q').Bool()
 	maxBlockDuration := importCmd.Flag("max-block-duration", "Maximum duration created blocks may span. Anything less than 2h is ignored.").Hidden().PlaceHolder("<duration>").Duration()
 	openMetricsImportCmd := importCmd.Command("openmetrics", "Import samples from OpenMetrics input and produce TSDB blocks. Please refer to the storage docs for more details.")
+	openMetricsLabels := openMetricsImportCmd.Flag("label", "Label to attach to metrics. Can be specified multiple times.").StringMap()
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply cmd/promtool/main.go:255]: Hello @machine424, Thanks for the review and sorry for the late reply.
I haven't used kingpin before so not completely sure why I need `Arg`, As per the kingpin doc `Arg` is used to provide argument to the subcommands. I have used `.Flag` in this case `openMetricsLabels` itself will hold the labels in form of `map[string]string`. Am I missing something here? 
Also after adding the above piece of code. I tried the below command and it is working as expected. 
```
promtool tsdb create-blocks-from openmetrics --label=cluster_name=demo --label=user=test /path/to/file
```

Regarding the unit test, I have added it in my latest commit.

[the remark this answers is not in the document]
Comment (machine424) [reply cmd/promtool/main.go:255]: Thanks for the tests.
You're right about the flag, we already have a similar `--label` one here https://prometheus.io/docs/prometheus/latest/command-line/promtool/#promtool-push

File: cmd/promtool/backfill_test.go [hunk 3]
```diff
@@ -92,6 +92,7 @@ func TestBackfill(t *testing.T) {
 		Description          string
 		MaxSamplesInAppender int
 		MaxBlockDuration     time.Duration
+		labels               map[string]string
```
Comment (machine424) [inline cmd/promtool/backfill_test.go:95]: To keep coherence
```suggestion
		Labels               map[string]string
```

File: cmd/promtool/backfill.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (machine424) [reply cmd/promtool/backfill.go:165]: Maybe we could re-use the same builder.
@bboreham found it more efficient in another use case here https://github.com/prometheus/prometheus/pull/13491

File: cmd/promtool/backfill_test.go (hunk 3 above)
Comment (patilsuraj767) [reply cmd/promtool/backfill_test.go]: Done

File: docs/command-line/promtool.md [hunk 4]
```diff
@@ -632,6 +632,15 @@ Import samples from OpenMetrics input and produce TSDB blocks. Please refer to t
 
 
 
+###### Flags
+
+| Flag | Description |
+| --- | --- |
+| <code class="text-nowrap">--label</code> | Label to attach to metrics. Can be specified multiple times. |
```
Comment (bboreham) [inline docs/command-line/promtool.md:639]: Can we show the name=value syntax?  I don't think people will guess.
Like you did in the PR description - `--label=cluster_name=demo --label=user=test`

Comment (dgl) [reply docs/command-line/promtool.md:639]: Note the docs are generated, so change in the flag description in the code.

Comment (patilsuraj767) [reply docs/command-line/promtool.md]: @bboreham  Added the `--label` example in the help command. 
@dgl Yes, I am auto generating the docs using `make cli-documentation`

**Review by machine424 — APPROVED** [review]
lgtm
Thanks again!

**Comment by dgl** [comment]
Thanks!

**Pull request merged**